### PR TITLE
For expr `return (_ = 42);` unused_paren lint should not be triggered

### LINKED ISSUE #131989: False positive for `unused_parens` in a `return`
### Code

```rust
fn foo() {
    return (_ = 42);
}
```

### Current output

```text
warning: unnecessary parentheses around `return` value
 --> src/lib.rs:2:12
  |
2 |     return (_ = 42);
  |            ^      ^
  |
  = note: `#[warn(unused_parens)]` on by default
help: remove these parentheses
  |
2 -     return (_ = 42);
2 +     return _ = 42;
  |
```

### Desired output

No output, since removing the parentheses results in a compile error:

```text
error: expected one of `;`, `}`, or an operator, found reserved identifier `_`
 --> src/lib.rs:2:12
  |
2 |     return _ = 42;
  |            ^ expected one of `;`, `}`, or an operator
```

### Rationale and extra context

_No response_

### Other cases

_No response_

### Rust Version

```console
$ rustc -vV
rustc 1.84.0-nightly (662180b34 2024-10-20)
binary: rustc
commit-hash: 662180b34d95f72d05b7c467b0baf4d23d36b1e1
commit-date: 2024-10-20
host: x86_64-unknown-linux-gnu
release: 1.84.0-nightly
LLVM version: 19.1.1
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by surechen**
fixes #131989

### CODE REVIEW

File: compiler/rustc_lint/src/unused.rs [hunk 1]
```diff
@@ -1063,6 +1063,8 @@ impl UnusedDelimLint for UnusedParens {
                                 _,
                                 _,
                             ) if node.is_lazy()))
+                    && !(ctx == UnusedDelimsCtx::ReturnValue
```
Comment (Nadrieril) [inline compiler/rustc_lint/src/unused.rs:1066]: Presumably a similar error would happen with `break`, does that capture it?

File: tests/ui/lint/unused/unused-parens-assign-expr-in-ret-issue-131989.rs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+//@ check-pass
+#![warn(unused_parens)]
```
Comment (Nadrieril) [inline tests/ui/lint/unused/unused-parens-assign-expr-in-ret-issue-131989.rs:2]: Could you set this to `deny` to be clearer about intent?

File: compiler/rustc_lint/src/unused.rs (hunk 1 above)
Comment (Nadrieril) [reply compiler/rustc_lint/src/unused.rs:1066]: Looking at the code, it looks like `ExprKind::Break` isn't treated specially. Would it be worth adding a variant to `UnusedDelimsCtx`?

Comment (surechen) [reply compiler/rustc_lint/src/unused.rs]: > Looking at the code, it looks like `ExprKind::Break` isn't treated specially. Would it be worth adding a variant to `UnusedDelimsCtx`?

Thank you. I add BreakValue to UnusedDelimsCtx. Please review again.

**Comment by Nadrieril** [comment]
Looks good, thank you!

@bors r+

**Pull request merged**